### ISSUE TITLE
Show metadata in ImageView

### ISSUE THREAD

**Issue by ukleinek** [k3b/APhotoManager#70]
I'd like to have a possibility to show a caption for the photos viewed in A Photo Manager. It might be possible that this already works and I just missed the relevant settings item as I'm new to this app.

I set various fields in the metadata of a picture using digikam, but I only found them in the "details" view which is way to verbose to look at while browsing the photos of the last vacation.
### Environment
- Android Version 4.4.4 (CyanogenMod 11 20150901) on Samsung Galaxy S
- A Photo Manager Version 0.5.2 160726 (via F-Droid)

**Comment by k3b** [comment]
"caption for the photos viewed in A Photo Manager".

I assume you mean showing the info in the gallery view and not in the image detail-view.

Which fields do you want to see  ? currently you see the end of the filename in the gallery-view.

This would require to implement a different gallery listview with some text columns.

Related #50: define the number of imagecolums.

**Comment by ukleinek** [comment]
I'd like to have a possibility to show a caption, a oneline-title would be fine. The source could be Exif Image Description or what is shown as "UserComment" in the details list.

For the question where to show it: I'd like to see it at least in the "show only one image" view (detail view?) to substitute printed photo books.

**Comment by k3b** [comment]
I have to check if the android mediascanner (on my "wiko-lenny-android-4.4" and on my "android-4.2 tablet") reads the "Exif Image Description" field into the media database. "UserComment" is not supported by android-stock-database/mediascanner/exif-interface" so this field has to wait until #25 is finished. the mediadb has a field "description" but for all of my 16000 images it-s value is null

**Comment by ukleinek** [comment]
Do I have the same mediadb? Can I easily check the description field for a given photo?

**Comment by ukleinek** [comment]
I found in the adb shell:

```shell
    # sqlite3 ./data/com.android.providers.media/databases/external.db
    sqlite> select * from images where not description is null;
    sqlite> 
```
though also at least one of my pictures has meta data set using digikam and the image is referenced in the database.

**Comment by keunes** [comment]
The meta-data I would like to be able to see is the creation date & time.

> This would require to implement a different gallery listview with some text columns.

I would rather keep the current view but have a setting to choose which data to display File name (current), date & time stamp, 'title' (as proposed by @ukleinek).

**Comment by k3b** [comment]
With progess beeing made for #25 "Tag support" i found a way to store and retrieve additional data in the android stock media-db so in future additional image meta data will be available but that-s a huge amount of work ahead, since this is a hobby project.

.h2 todo:

* either find an android lib to read/write exif/iptc/xmp/jpg meta data 
  * or go on implementing my own https://github.com/k3b/pixymeta-android
* integrate it into aPhotoManager-media scanner
* allow filtering for the additional data
* create a listview with more text content (including UserComment, date, filename without path, title)
* import/export exiftool compatible csv to speedup inital scan

**Comment by Rbagman** [comment]
I would simply like the ability to edit exif data; I haven't found one F-Droid app that can do this. 

Specifically, two options:

A quick Privacy Wipe option.

A fully realized exif editor.

Thanks

**Comment by keunes** [comment]
> I would simply like the ability to edit exif data; I haven't found one F-Droid app that can do this. Specifically, two options:
A quick Privacy Wipe option.
A fully realized exif editor.

That's much more than what this feature request is about and involves a lot of different work. So it'd be best to open a new feature request for this...

**Comment by k3b** [comment]
@Rbagman 

> A quick Privacy Wipe option.

You can use "ObsucraCam" from fdroide-store "the Guardian Project Official Releases" which hooks into the "send to" functiion to remove exif-infos before sending an image.

**Comment by Rbagman** [comment]
Thanks. I've had this app for a while but it never completes 'Saving...'. However, I sent an attached image to myself and the app did wipe the exit data.

-----Original Message-----

**Comment by k3b** [comment]
Implementation is still under way. 

In version 0.6.1 the [Image-View](https://github.com/k3b/AndroFotoFinder/wiki/Image-View) 
will get **context text** and a **context button**.

Context means text and button relates to the last menu command you execute.

Example:

* if you called the menu command "Set geo" the context text will show latitude, longitude of the image and the context button will open the "Set geo" command.
* if you called the menu command "Set tags" the context text will show tags of the image and the context button will open the "Set tags" command.
* if you called the menu command "Rename" the context text will show file path of the image and the context button will open the "Rename" command.
* if you called the menu command "Details" the context text will show image-id, file path, lat/lon, tags, title, description, rating of the image and the context button will open the "Details" command.

If you know SQL and the properties file format you can define context specific sql-expressions for the **context text** using a text editor.

**Comment by k3b** [comment]
... will be available with next release 0.6.1
